### DISCUSSION TITLE
Can the sidebar be set to open and close the button?

### DISCUSSION THREAD

**Discussion by dongrentianyu** [web-infra-dev/rspress#491] [Q&A]
类似Docusaurus那样，可以通过按钮来开闭。不然的话一直放着或者直接取消都不太好。

**Comment by dongrentianyu** [comment]
不对，应该是说导航栏，说错了。
---

靠，好像又说错了，是文档左边的那部分内容。

**Reply by sanyuan0704** [reply]
这是一个不错的 feature，后续在默认主题加上隐藏/显示整体侧边栏的功能哈，参考：
- Docusaurus: https://docusaurus.io/docs/category/getting-started
- Nextra: https://nextra.site/docs

**Reply by dongrentianyu** [reply]
感谢。这个功能还挺好的。然后另一边文章的大纲如果也能够开闭就好了。因为有时候需要大纲，有时候又不需要大纲。有个开闭的按钮，能更好展示效果。

**Reply by sanyuan0704** [reply]
没问题，都可以开闭，后续可以在主题加上

**Reply by dongrentianyu** [reply]
感谢感谢。

**Reply by sanyuan0704** [reply]
实际上也可以通过 frontmatter 来进行局部组件的开关控制 https://rspress.dev/zh/guide/basic/custom-page.html#%E4%BD%BF%E7%94%A8%E7%BB%86%E7%B2%92%E5%BA%A6%E5%BC%80%E5%85%B3

**Reply by dongrentianyu** [reply]
目前是采用这种方式。我基本上都关闭了。但有些时候还是需要打开。就不想每个mdx文件都处理开闭。当然应该还是frontmatter比控制按钮优先级要高。
